Mathlib/Data/Nat/Digits/Lemmas.lean

```diff
@@ -57,6 +57,27 @@ theorem digits_len (b n : ℕ) (hb : 1 < b) (hn : n ≠ 0) : (b.digits n).length
     contrapose! h
     exact div_eq_of_lt h
 
+theorem digits_length_le_of_lt {b k : ℕ} (hb : 1 < b) (n : ℕ) (hbn : n < b ^ k) :
+    (b.digits n).length ≤ k := by
+  by_cases h : n = 0
+  · simp [h]
+  rw [digits_len b n hb h]
+  rw [lt_pow_iff_log_lt hb h] at hbn
+  exact hbn
+
+theorem digits_length_le_iff {b k : ℕ} (hb : 1 < b) (n : ℕ) :
```

Comment:
Since the conditions are the same, I'd argue you should just inline the `of` lemma into the `iff` one.